### ISSUE TITLE
Cluster management functionality

### ISSUE THREAD

**Issue by vitabaks** [autobase-tech/autobase#754]
With the release version [2.0](https://github.com/vitabaks/postgresql_cluster/releases/tag/2.0.0), we introduced a Console (UI) designed for deploying clusters through a graphical interface. 

![Image](https://github.com/user-attachments/assets/cf72f095-5bcb-47e8-8a49-f980a273d9f7)

The next step is to add functionality for managing an already deployed cluster through the UI, including:

- [x] **Edit** a cluster
  - Change the cluster name, description, environment, synchronous mode, etc.
- [x] **Stop**/**Start** a cluster or a specific cluster node
  - Control the Patroni service or the VM (for cloud deployments)
- [x] **Switchover**/**Failover**, **Reinit**, **Reload**, **Restart**
  - Integration with the Patroni [API](https://patroni.readthedocs.io/en/latest/rest_api.html)
- [x] **Users**
  - Create and edit database users
- [x] **Databases**
  - Create and edit databases
- [x] **Extensions**
  - Create and edit database extensions
- [x] **Parameters**
  - Edit Postgres and system parameters
  - Manage connection settings
    - Enable/disable public access to the database (for cloud deployments)
    - Configure connection pools
    - Manage pg_hba rules
- [x] **Backups**
  - Manage database backup settings
      - Backup schedule
      - Backup retention
      - Backup tool configuration
- [x] **Restore**
  - Restoring (PITR) or Cloning a cluster from a backup
- [x] **Upgrades**
  - Minor and major cluster upgrade
- [x] **Scaling**
  -  Scaling the cluster - add Postgres node
- [x] **Delete** a cluster
  - Remove cluster components from the server or the entire VM (for cloud deployments)

> [!NOTE]
> Access to the Figma project with all implementation details will be provided during development.

---
Consider sponsoring the project via [GitHub](https://github.com/sponsors/vitabaks) or [Patreon](https://patreon.com/vitabaks)

**Comment by vitabaks** [comment]
If you're interested in this feature, please consider becoming a [sponsor](https://github.com/vitabaks/postgresql_cluster?tab=readme-ov-file#sponsor-this-project).

The development of this feature requires sponsorship to fund developer efforts. If you're already a sponsor and are interested in this feature, please leave a comment here so we can prioritize this issue accordingly.

We also welcome contributions from those willing to dedicate their time to implement this feature on a voluntary basis. Please see the [Contributing Guide](https://github.com/vitabaks/postgresql_cluster/blob/master/CONTRIBUTING.md) for more information.

**Comment by stanlog** [comment]
Hello @vitabaks thank you so much for this marvelous software, and all this work of automation with Ansible.

I don't know if it's the right place to put a wish list, but I realized a little lack last days when testing deployments : I can't import an existing cluster deployed through command line and Ansible into the UI (or I pass by this feature). Do you think it would be possible ? (even if I have to connect in bash in the container and launch several commands to do the job). I tried to deploy my cluster in the UI, adapted vars in Ansible automation files, and config / update cluster launching ansible-playbook commands, but it's not really simple, and not really do the job in some cases.

**Comment by vitabaks** [comment]
Hello @stanlog 
Thank you for your feedback and suggestion.

For the feature request to import an existing cluster into the Console, it’s better to create a separate issue so we can track and consider it for future development.

In the meantime, this can be done manually by adding the cluster data directly into the [clusters](https://github.com/vitabaks/postgresql_cluster/blob/master/console/db/migrations/20240520144338_2.0.0_initial_scheme_setup.sql#L605) and [servers](https://github.com/vitabaks/postgresql_cluster/blob/master/console/db/migrations/20240520144338_2.0.0_initial_scheme_setup.sql#L671) tables in the console database.

**Comment by stanlog** [comment]
> For the feature request to import an existing cluster into the Console, it’s better to create a separate issue so we can track and consider it for future development.

Yes, I will create a dedicated issue to ask for this enhancement.

Thank you for the workaround to populate the database in the UI to import an existing cluster :)

**Comment by vitabaks** [comment]
In progress.

**Comment by KazimirPodolski** [comment]
Is this released already? I don't see any of those options in my UI.

**Comment by vitabaks** [comment]
> Is this released already? I don't see any of those options in my UI.

Most of the functions have already been implemented; they are included in the [Autobase Enterprise](https://autobase.tech/docs#getting-started) edition. We are currently working on the 2.11 release, in which we plan to add all the remaining management functionality.

**Comment by KazimirPodolski** [comment]
> they are included in the [Autobase Enterprise](https://autobase.tech/docs#getting-started) edition

I wish there was a tier between Free and Standard. I don't need the DBA hour nor the SLA (IMO neither is really expected when self-hosting), but I can use full functionality of the console. Especially how essential it is really. On the other hand, I cannot afford Standard price right now - it alone will be 2.5x of my whole infra costs at the moment, and the project doesn't even make a revenue yet.

**Comment by vitabaks** [comment]
@KazimirPodolski Unfortunately, we don’t plan to include these features in the Free edition, which is primarily intended for deployment. Advanced management, monitoring, RBAC, and future Enterprise features are part of the paid platform.

We understand the concern about pricing, but maintaining and developing a project of this scale requires significant resources and a sustainable business model.

**Comment by pavelkovcheg** [comment]
I get the pricing concern, but there’s one small problem: building and maintaining Autobase costs money 😄

Developers need at least some bread. Then, unexpectedly, there are also children, rent, electricity, and internet bills.

Of course, we could replace laptops with a "starry sky, meditation, and strong belief in open source".

Sadly, servers, engineers, and infrastructure providers still refuse to accept inner peace as payment 🙂

**Comment by KazimirPodolski** [comment]
Of course, nobody says it should be free. It's just currently I don't see how Autobase could be viable for small time companies / bootstrapping startups. Either I don't get something, or you are missing out on this market because there is no suitable tier.

**Comment by KazimirPodolski** [comment]
Btw, Trial version does NOT have those features either, despite the website claims it's an Enterprise Edition.

**Comment by vitabaks** [comment]
> Btw, Trial version does NOT have those features either, despite the website claims it's an Enterprise Edition.

Perhaps you tried to upgrade the community edition to the enterprise edition; try a new installation instead, as they have different versions of the Postgres schema.

**Comment by vitabaks** [comment]
> It's just currently I don't see how Autobase could be viable for small time companies / bootstrapping startups

However, command-line and GitOps-based management are available completely free of charge. You can start with these and switch to the full Enterprise version when needed.

**Comment by vitabaks** [comment]
Done. The management functionality is available in [Autobase Enterprise](https://autobase.tech/docs#getting-started).
